src/testing/fuzz.zig

```diff
@@ -88,6 +88,29 @@ pub fn parse_seed(bytes: []const u8) u64 {
     };
 }
 
+// Like `std.meta.DeclEnum`, but allows excluding specific things. Feed the result into
+// random_enum_weights for swarm testing public API of a data structure.
+pub fn DeclEnumExcludingType(T: type, exclude: []const std.meta.DeclEnum(T)) type {
+    const base = @typeInfo(std.meta.DeclEnum(T)).Enum;
+    var fields_filtered: [base.fields.len - exclude.len]std.builtin.Type.EnumField = undefined;
+    var i: usize = 0;
+    next_field: for (base.fields) |field| {
+        for (exclude) |excluded| {
+            if (std.mem.eql(u8, field.name, @tagName(excluded))) continue :next_field;
+        }
```

Comment:
TIL it's valid in Zig 0.13 :heart:

Reply:
My intuition would tell me the outer `for` loop would start at `0` again! But probably I'm mixing it with the new labelled continue on `switch` introduced in Zig 0.14.

Reply:
Yeah, no, that's just standard labeled continue:

https://docs.oracle.com/javase/tutorial/java/nutsandbolts/branch.html#:~:text=A%20labeled%20continue

I tried rewriting it as

```zig
    for (base.fields) |field| {
        const field_excluded = for (exclude) |excluded| {
            if (std.mem.eql(u8, field.name, @tagName(excluded))) break true;
        } else false;
        if (!field_excluded) {
            fields_filtered[i] = field;
            i += 1;
        }
    }
```

But I don't really love how we flip modality here.

Reply:
Related post I always think about when writing loops:

https://www.teamten.com/lawrence/programming/avoid-continue.html